2011-06-29  Michael Tautschnig  <mt@debian.org>

	* setup-storage/Sizes.pm: Use (virtual) end of disk for 100% size computation
		instead of disk size to fix 100% and gpt configuration issue
		(closes: #619136).
Index: trunk/lib/setup-storage/Sizes.pm
===================================================================
--- trunk.orig/lib/setup-storage/Sizes.pm
+++ trunk/lib/setup-storage/Sizes.pm
@@ -460,7 +460,7 @@
     $next_start;
 
   if (1 == $part_id) {
-    $max_avail = $current_disk->{size} - $next_start;
+    $max_avail = $current_disk->{end_byte} + 1 - $next_start;
     $max_avail = "${max_avail}B";
   }
   my ($start, $end) = &FAI::make_range($part->{size}->{range}, $max_avail);
@@ -671,7 +671,7 @@
     }
 
     # the size of a 100% partition (the 100% available to the user)
-    my $max_avail = $current_disk->{size} - $next_start;
+    my $max_avail = $current_disk->{end_byte} + 1 - $next_start;
     # expressed in bytes
     $max_avail = "${max_avail}B";
 
